pkg/failover/server.go

```diff
@@ -72,12 +72,25 @@ func (s *Server) NewAuthenticationInterceptor(token string) func(context.Context
 func (s *Server) HealthCheck(ctx context.Context, _ *Empty) (*HealthCheckResponse, error) {
 	resp := &HealthCheckResponse{Status: HealthCheckResponse_HEALTHY}
 
-	// TODO: Provide error in the health check response
-	// https://github.com/gocardless/stolon-pgbouncer/pull/11
+	// PgBouncer Healthcheck
+	pgBouncerHealthCheck := &HealthCheckResponse_ComponentHealthCheck{
+		Name:   "PgBouncer",
+		Status: HealthCheckResponse_HEALTHY,
+	}
+
 	if _, err := s.bouncer.ShowDatabases(ctx); err != nil {
-		resp.Status = HealthCheckResponse_UNHEALTHY
+		pgBouncerHealthCheck.Status = HealthCheckResponse_UNHEALTHY
+		pgBouncerHealthCheck.Error = err.Error()
 	}
 
+	resp.Components = []*HealthCheckResponse_ComponentHealthCheck{pgBouncerHealthCheck}
+
+	// We only have one component right now, so the overall status == the status of that
```

Comment:
Is there a point in having this comment or can we just write the code as if there were multiple components? Doesn't make it much more complex otherwise and it removes a trap for anyone who adds another component later

Reply:
I think it depends what kind of component you add in the future - e.g. you might have an optional healthcheck that doesn't affect the overall result. I didn't want to attempt to guess how this might evolve in the future.

Reply:
Is it worth commenting it then? Either way it's second guessing

Reply:
I thought that was communicated by this comment, to be honest. All I've said here is this is how this currently works, and you might want to take this approach when you add a new component.